lib/manager/bundler/extract.js

```diff
@@ -1,159 +1,155 @@
-      const sourceRes = await extractPackageFile(sourceContent);
-      if (sourceRes) {
-        res.deps = res.deps.concat(
-          sourceRes.deps.map(dep => ({
-            ...dep,
-            registryUrls: [repositoryUrl],
-            lineNumber: dep.lineNumber + sourceLineNumber + 1,
-          }))
-        );
-      }
-    }
-    const platformsMatch = line.match(/^platforms\s+(.*?)\s+do/);
-    if (platformsMatch) {
-      const platformsLineNumber = lineNumber;
-      let platformsContent = '';
-      let platformsLine = '';
-      while (lineNumber < lines.length && platformsLine !== 'end') {
-        lineNumber += 1;
-        platformsLine = lines[lineNumber];
-        if (platformsLine !== 'end') {
-          platformsContent += platformsLine.replace(/^ {2}/, '') + '\n';
-        }
-      }
-      const platformsRes = await extractPackageFile(platformsContent);
-      if (platformsRes) {
-        res.deps = res.deps.concat(
-          // eslint-disable-next-line no-loop-func
-          platformsRes.deps.map(dep => ({
-            ...dep,
-            lineNumber: dep.lineNumber + platformsLineNumber + 1,
-          }))
-        );
-      }
-    }
-    const ifMatch = line.match(/^if\s+(.*?)/);
-    if (ifMatch) {
-      const ifLineNumber = lineNumber;
-      let ifContent = '';
-      let ifLine = '';
-      while (lineNumber < lines.length && ifLine !== 'end') {
-        lineNumber += 1;
-        ifLine = lines[lineNumber];
-        if (ifLine !== 'end') {
-          ifContent += ifLine.replace(/^ {2}/, '') + '\n';
-        }
-      }
-      const ifRes = await extractPackageFile(ifContent);
-      if (ifRes) {
-        res.deps = res.deps.concat(
-          // eslint-disable-next-line no-loop-func
-          ifRes.deps.map(dep => ({
-            ...dep,
-            lineNumber: dep.lineNumber + ifLineNumber + 1,
-          }))
-        );
-      }
-    }
-  }
-  if (!res.deps.length && !res.registryUrls.length) {
-    return null;
-  }
-  if (fileName) {
-    const gemfileLock = fileName + '.lock';
-    const lockContent = await platform.getFile(gemfileLock);
-    if (lockContent) {
-      logger.debug({ packageFile: fileName }, 'Found Gemfile.lock file');
-      const bundledWith = lockContent.match(/\nBUNDLED WITH\n\s+(.*?)(\n|$)/);
-      if (bundledWith) {
-        res.compatibility = res.compatibility || {};
-        res.compatibility.bundler = bundledWith[1];
-      }
-    }
+};
+
+const createTempfile = async ({ content, name }) => {
+  const tmpPath = tmpdir();
+  const filePath = join(tmpPath, name);
+
+  await outputFile(filePath, content);
+
+  return filePath;
+};
+
+const createLogger = level => curry(logger[level], 2);
+
+const createParser = type => async path => {
+  const shell = true;
+  const cwd = __dirname;
+  const bin = './parser/parse';
+
+  const args = getCmdArgs([type, resolve(path)]);
+  const env = getCmdEnv(global.trustLevel);
+  const { stdout } = await exec(`${bin} ${args}`, { cwd, shell, env });
```

Comment:
TODO: if binarySource === 'docker' then call out to Docker rather than the code directly